### PR TITLE
Fix InlinedVector to use its elements' move and copy methods.

### PR DESCRIPTION
**Pull request by markdroth**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%     +16 [None]                                                                               +7.72Ki  +0.1%

 -------------- SHRINKING                                                                            --------------
  -1.7%    -112 src/core/ext/filters/client_channel/resolver_result_parsing.cc                          -112  -1.7%
     -10.0%    -112 bool grpc_core::ServiceConfig::ParseJsonMethodConfig<grpc_core::internal::ClientChan    -112 -10.0%
  -1.6%     -80 src/core/ext/filters/message_size/message_size_filter.cc                                 -80  -1.6%
      -2.9%     -72 init_channel_elem                                                                        -72  -2.9%
      -6.3%      -8 [Unmapped]                                                                                -8  -6.3%
  -6.6%     -64 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_client_stats.cc              -64  -6.6%
      -8.9%     -62 grpc_core::GrpcLbClientStats::AddCallDroppedLocked                                       -62  -8.9%
      -7.1%      -2 [Unmapped]                                                                                -2  -7.1%
  -0.5%     -64 src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc                   -64  -0.5%
      -1.0%     -32 grpc_core::(anonymous namespace)::PickFirst::UpdateLocked                                -32  -1.0%
      -5.7%     -16 [Unmapped]                                                                               -16  -5.7%
      -0.8%      -8 grpc_core::(anonymous namespace)::PickFirst::UpdateChildRefsLocked                        -8  -0.8%
      -1.7%      -8 grpc_core::(anonymous namespace)::PickFirst::FillChildRefsForChannelz                     -8  -1.7%
  -0.5%     -64 src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.cc                 -64  -0.5%
      [DEL]    -592 grpc_core::(anonymous namespace)::RoundRobin::RoundRobinSubchannelList::StartWatchin    -592  [DEL]
      -2.3%     -64 grpc_core::(anonymous namespace)::RoundRobin::UpdateLocked                               -64  -2.3%
      -6.5%     -16 [Unmapped]                                                                               -16  -6.5%
      -1.7%      -8 grpc_core::(anonymous namespace)::RoundRobin::FillChildRefsForChannelz                    -8  -1.7%
  -6.6%     -64 src/core/ext/filters/client_channel/lb_policy/xds/xds_client_stats.cc                    -64  -6.6%
      -8.9%     -62 grpc_core::XdsLbClientStats::AddCallDroppedLocked                                        -62  -8.9%
      -7.1%      -2 [Unmapped]                                                                                -2  -7.1%
  -4.9%     -64 src/core/ext/filters/client_channel/lb_policy_registry.cc                                -64  -4.9%
     -11.6%     -64 grpc_core::LoadBalancingPolicyRegistry::Builder::RegisterLoadBalancingPolicyFactory      -64 -11.6%
  -2.0%     -64 src/core/ext/filters/client_channel/resolver_registry.cc                                 -64  -2.0%
     -11.6%     -64 grpc_core::ResolverRegistry::Builder::RegisterResolverFactory                            -64 -11.6%
  -0.8%     -48 src/core/lib/iomgr/udp_server.cc                                                         -48  -0.8%
      -3.4%     -46 add_socket_to_server                                                                     -46  -3.4%
      -1.4%      -2 [Unmapped]                                                                                -2  -1.4%
  -0.0%     -16 src/core/ext/filters/client_channel/client_channel.cc                                    -16  -0.0%
      [DEL]    -397 add_closure_for_subchannel_batch(grpc_call_element*, grpc_transport_stream_op_batch*    -397  [DEL]
      [DEL]    -192 add_retriable_send_message_op(grpc_call_element*, (anonymous namespace)::subchannel_    -192  [DEL]
      -0.8%     -17 run_closures_for_completed_call                                                          -17  -0.8%
  -0.1%     -16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                           -16  -0.1%
      [DEL] -1.97Ki grpc_core::(anonymous namespace)::GrpcLb::CreateOrUpdateRoundRobinPolicyLocked() [cl -1.97Ki  [DEL]
      -2.4%      -9 [Unmapped]                                                                                -9  -2.4%
      -2.5%      -7 grpc_core::(anonymous namespace)::GrpcLb::FillChildRefsForChannelz                        -7  -2.5%
  -0.1%     -16 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                 -16  -0.1%
      [DEL] -1.81Ki grpc_core::(anonymous namespace)::XdsLb::CreateOrUpdateChildPolicyLocked() [clone .p -1.81Ki  [DEL]
      -2.4%      -9 [Unmapped]                                                                                -9  -2.4%
      -2.5%      -7 grpc_core::(anonymous namespace)::XdsLb::FillChildRefsForChannelz                         -7  -2.5%
  -0.7%     -16 src/core/lib/transport/transport.cc                                                      -16  -0.7%
      -1.2%     -16 grpc_transport_stream_op_batch_finish_with_failure                                       -16  -1.2%

  -0.0%    -672 TOTAL                                                                                +7.05Ki  +0.1%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +24  +0.0%

  [ = ]       0 TOTAL      +24  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/lib/gprpp/inlined_vector.h [hunk 1]
```diff
@@ -158,17 +168,13 @@ class InlinedVector {
...
-
-  size_t capacity() const { return capacity_; }
-
-  void clear() {
-    destroy_elements();
-    init_data();
+  static void move_elements(T* src, T* dst, size_t num_elements) {
+    for (size_t i = 0; i < num_elements; ++i) {
+      new (&dst[i]) T(std::move(src[i]));
+      src[i].~T();
```
Comment (ncteisen) [inline src/core/lib/gprpp/inlined_vector.h]: Is this needed? Won't this be accomplished from the move operation?

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,244      Total (>)      2,020,224

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
 3,949,226       Core (>)      3,947,490

11,155,967      Total (>)     11,154,228


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by markdroth** [comment]
Known issues: #17335 #17336

**Pull request merged**